src/vaadin-text-field-mixin.html

```diff
@@ -450,32 +450,20 @@
     }
 
     _labelChanged(label) {
-      if (label !== '' && label != null) {
-        this.setAttribute('has-label', '');
-      } else {
-        this.removeAttribute('has-label');
-      }
+      this._setOrToggleAttribute('has-label', label === 0 ? true : label, this);
```

Comment:
label is of type string so `label === 0` can never be true unless a developer illegally assigns a value of wrong type

Reply:
I had to check for that scenario because of [this](https://github.com/vaadin/vaadin-text-field/blob/f5cc5da639abaed7f5145a00958bf7a910118b8a/test/text-field.html#L301) test.

Reply:
:) Oh lord. We actually have tests to verify that the property works with invalid value types  🤦‍♂ ...while the property is even documented to be of type String. I guess it's safer now to keep the current behaviour. This will get fixed when we switch to TypeScript.

Reply:
It could be shortened to `label === 0 || !!label` however